components/scripts/lib/build-scan-dump.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+readonly BUILD_SCAN_SUPPORT_TOOL_JAR="${SCRIPT_DIR}/build-scan-support-tool.jar"
+
+build_scan_dumps=()
+
+verify_build_scan_support_tool_exists() {
+  if [ ! -f "$BUILD_SCAN_SUPPORT_TOOL_JAR" ]; then
+    die "ERROR: build-scan-support-tool.jar is required when using --disable-build-scan-publishing."
+  fi
+}
+
+find_and_read_build_scan_dumps() {
+  find_build_scan_dumps
+  read_build_scan_dumps
+}
+
+find_build_scan_dumps() {
+  find_build_scan_dump "first"
+  find_build_scan_dump "second"
+}
+
+find_build_scan_dump() {
+  local build_name="$1"
+  local build_dir="${EXP_DIR}/$build_name-build_${project_name}"
+  if [ -n "${project_dir}" ]; then
+    build_dir="$build_dir/$project_dir"
+  fi
+  build_scan_dump="$(find "$build_dir" -maxdepth 1 -type f -regex '^.*build-scan-.*-.*-.*-.*\.scan' | sort | tail -n 1)"
+  if [ -z "$build_scan_dump" ]; then
+    die "ERROR: No Build Scan dump found for the $build_name build"
+  fi
+  build_scan_dumps+=("${build_scan_dump}")
+}
+
+read_build_scan_dumps() {
+  local build_scan_csv
+  echo -n "Extracting build scan data"
+  build_scan_csv="$(invoke_java "$BUILD_SCAN_SUPPORT_TOOL_JAR" extract "${build_scan_dumps[0]}"  "${build_scan_dumps[1]}")"
+  parse_build_scan_csv "$build_scan_csv"
+  echo ", done."
+}
```

Comment:
It seems not ideal to me that we echo `Extracting build scan data"` in this helper methods. We don't do it when reading from GE neither. Can the echoing be moved to the invoker? Would this also make it more symmetric compared to reading from the GE API?

Reply:
This was originally done to avoid having to change build-scan-support-tool to print this message out, since we had limited time before the DPE Summit presentation.

But now we have more time, so I think we should move this to build-scan-support-tool.

Reply:
Thanks for hopping in @jthurne! 

@etiennestuder would you like me to leave or remove this? Let me know if you'd like me to create a follow up ticket to update the `build-scan-support-tool.jar`.

cc @sorin-florea

Reply:
Let's leave it for now. While not looking great in the code, IMO, it works. We have bigger things to tackle first.